test/libyul/yulOptimizerTests/fullSuite/unusedFunctionParameterPruner_loop.yul

```diff
@@ -19,19 +19,47 @@
 //
 // {
 //     {
+//         f_63()
 //         f()
-//         f()
-//         f()
+//         f_65()
```

Comment:
This looks like a regression compared to the default sequence. `UnusedFunctionParameterPruner` (`p`) does not remove the obviously unnecessary parameter so the function gets specialized. It happens because when we run `p`, the body contains an (unused) assignment to the parameter:

```yul
        function f(a) -> x {
            (...)
            for { } true { (...) } {
                (...)
                a := a_1
                mstore(a_1, 0)
                (...)
            }
        }
```

The assignment is created by the SSA but not removed by `UnusedAssignEliminator` (`r`) that runs afterwards because the variable is used as initialization for some (unused) variables at that point:
```yul
        function f(a) -> x {
            (...)
            for { } true { (...) let a_2 := a (...) } {
                (...)
                let a_3 := a
                (...)
                a := a_4
                (...)
            }
        }
```

This can be solved by either replacing the `[r]` loop after the SSA with `[ru]` or by running `r` before `p`.

I think that the default sequence does not have this problem due to the presence of the simplification segment, which includes `u` and runs before we reverse SSA.

Reply:
### Comparison with unmodified `the-good-parts-mk2` on `cancun`
#### Tweak
```diff
-       "Tpeul"                        // Run functional expression inliner
+       "Trpeul"                       // Run functional expression inliner
```

#### External tests (`ir-optimize-evm+yul`)
[`seqbench-sequence-the-good-parts-mk2-tweaks-r-before-p`](https://app.circleci.com/pipelines/github/ethereum/solidity/33656/workflows/18144d7d-10e6-40bd-9a59-ac37fb90c96f) vs [`seqbench-sequence-the-good-parts-mk2-cancun`](https://app.circleci.com/pipelines/github/ethereum/solidity/33655/workflows/f901ee55-094e-44b1-860c-44cab6b675bb)

|          project |  bytecode_size | deployment_gas |     method_gas |
|:----------------:|---------------:|---------------:|---------------:|
|        elementfi | **`-0.03% ✅`** |                |                |
|              ens | **`-0.01% ✅`** |           `0%` |           `0%` |
|            euler | **`-0.07% ✅`** |                |                |
|          uniswap | **`+0.35% ❌`** |                |                |
| yield_liquidator | **`-0.02% ✅`** | **`-0.03% ✅`** |          `-0%` |
|         zeppelin | **`-0.01% ✅`** | **`-0.01% ✅`** | **`+0.04% ❌`** |

#### Gas diff stats
No differences over +/-0.5%.